riffusion/streamlit/pages/audio_to_audio_batch.py

```diff
@@ -0,0 +1,221 @@
+    )
+
+    clip_start_time_ms = int(clip_start_time_s * 1000)
+    clip_duration_ms = int(clip_duration_s * 1000)
+    clip_segment: pydub.AudioSegment = segment[clip_start_time_ms: clip_start_time_ms + clip_duration_ms]
+
+    audio_bytes = io.BytesIO()
+    clip_segment.export(audio_bytes, format="wav")
+
+    params = SpectrogramParams()
+
+    init_image: Image.Image = streamlit_util.spectrogram_image_from_audio(
+        clip_segment,
+        params=params,
+        device=device,
+    )
+
+    closest_width = int(np.ceil(init_image.width / 32) * 32)
+    closest_height = int(np.ceil(init_image.height / 32) * 32)
+    init_image_resized = init_image.resize((closest_width, closest_height), Image.BICUBIC)
+
+    st.write("#### Source Clip")
+    with st.expander("Source Clip"):
+        st.audio(audio_bytes)
+        st.image(init_image, use_column_width=False)
+
+    st.write("#### Settings")
+    with st.form("Conversion Params"):
+
+        prompt = st.text_input("Text Prompt")
+        negative_prompt = st.text_input("Negative Prompt")
+
+        cols = st.columns(3)
+        denoising_strength = cols[0].number_input(
+            "Denoising Strength",
+            min_value=0.0,
+            max_value=1.0,
+            value=0.50,
+        )
+        guidance_scale = cols[1].number_input(
+            "Guidance Scale",
+            min_value=0.0,
+            max_value=20.0,
+            value=7.0,
+        )
+        num_inference_steps = int(
+            cols[2].number_input(
+                "Num Inference Steps",
+                min_value=1,
+                max_value=150,
+                value=50,
+            )
+        )
+
+        submit_button = st.form_submit_button("Convert", on_click=increment_counter)
+
+    if not prompt:
+        st.info("Enter a prompt")
+        return
+
+    if not submit_button:
+        return
+
+    for b in range(0, batches):
+        # Each batch has a random seed.
+        seed = randint(10, 100000)
+
+        container = st.container()
+        empty_bin = container.empty()
+        progress_callback = None
+
+        container.write(f"### Seed {seed}")
+
+        with empty_bin.container():
+            st.info(f"Riffing ({b}/{batches})...")
+            progress = st.progress(0.0)
+            progress_callback = progress.progress
+
+        left, right = container.columns(2)
+
+        left.write(f"##### Riffed Clip")
+
+        result_image: Image.Image = streamlit_util.run_img2img(
+            prompt=prompt,
+            init_image=init_image_resized,
+            denoising_strength=denoising_strength,
+            num_inference_steps=num_inference_steps,
+            guidance_scale=guidance_scale,
+            negative_prompt=negative_prompt,
+            seed=seed,
+            progress_callback=progress_callback,
+            device=device,
+        )
+
+        # Resize back to original size
+        result_image = result_image.resize(init_image.size, Image.BICUBIC)
```

Comment:
Does not attempt stitching as with `audio2audio`. The idea is that batching is more "experimental" to generate many short samples to assist dialling in prompts and finding good seed values, but then moving over to `audio2audio` for any extended track generation.